code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -363,4 +367,120 @@ static HookFunction initFunction([]()
 	{
 		*reinterpret_cast<float *>(wheelAddr + WheelXOffsetOffset) = context.GetArgument<float>(2);
 	}));
+
+	static struct : jitasm::Frontend
+	{
+		static bool ShouldSkipRepairFunc(fwEntity* VehPointer)
+		{
+			return g_skipRepairVehicles.find(VehPointer) != g_skipRepairVehicles.end();
+		}
+		virtual void InternalMain() override
+		{
+			//restore rcx after asm call
+			mov(rcx, rbx);
+			//save registers
+			push(rax);
+			push(rcx);
+			push(rdx);
+			push(r8);
+			push(r9);
+			push(r10);
+			push(r11);
+
+			sub(rsp, 0x28);
+			mov(rax, (uintptr_t)ShouldSkipRepairFunc);
+			call(rax);
+			add(rsp, 0x28);
+
+			//restore registers
+			pop(r11);
+			pop(r10);
+			pop(r9);
+			pop(r8);
+			pop(rdx);
+			pop(rcx);
+
+			test(al, al);
+			je("skiprepair");
+			pop(rax);
+			sub(rsp, 0x28);
+			AppendInstr(jitasm::InstrID::I_CALL, 0xFF, 0, jitasm::Imm8(2), qword_ptr[rax + 0x5D0]);
+			add(rsp, 0x28);
+			ret();
+			L("skiprepair");
+			pop(rax);
+			ret();
+		}
+	} asmfunc;
+
+	auto repairFunc = hook::get_pattern("48 8B 03 45 33 C0 B2 01 48 8B CB FF 90 D0 05 00 00 48 8B 4B 20",11);
+	hook::nop(repairFunc, 6);
+	hook::call_rcx(repairFunc, asmfunc.GetCode());
+
+	static struct : jitasm::Frontend
+	{
+		static void CleanupVehicle(fwEntity* VehPointer)
+		{
+			g_skipRepairVehicles.erase(VehPointer);
+		}
+		virtual void InternalMain() override
+		{
+			//restore rcx after asm call
+			mov(rcx, rdi);
```

Comment:
`rdi` isn't the place where the vehicle pointer is in all cases! there's other vehicle classes that _don't_ use rdi.

rdx is unused at this point, so you could use `call_reg<2>` or similar